### DISCUSSION TITLE
Support Supabase CLI Action in GitLab for Deployment

### DISCUSSION THREAD

**Discussion by AyoPrez** [supabase/supabase#13240] [Feature Requests]
Hi,
I find fantastic the work done with the Supabase CLI and the setup action for Github (https://github.com/supabase/setup-cli). It's probably not in the pipeline, but it would be awesome if you can support other CI/CD platforms. We personally use Gitlab, and it would save us, and probably others, a lot of headaches if we could have something like this ready to use.

**Comment by simpros** [comment]
Hi, did you make some progress on that topic? Did you try to get it running? 🤔

**Comment by AyoPrez** [comment]
I think someone replied this, but now it's not visible. I haven't had time to try it:

> You can configure GitLab CI/CD to deploy your application using the Supabase CLI action. Here's an example .gitlab-ci.yml configuration that demonstrates how you can use the Supabase CLI action to deploy your application to Supabase:
> .gitlab-ci.yml
> Define stages for the CI/CD pipeline
> 
> stages:
> 
>     build
>     deploy
> 
> Define jobs for each stage
> Example of a build job
> 
> build:
> stage: build
> script:
> - echo "Building the application..."
> # Run your build commands here
> Example of a deploy job
> 
> deploy:
> stage: deploy
> script:
> - echo "Deploying the application to Supabase..."
> # Install Supabase CLI
> - npm install -g supabase
> # Set Supabase environment variables
> - export SUPABASE_URL=<your_supabase_url>
> - export SUPABASE_KEY=<your_supabase_key>
> # Run the Supabase CLI action to deploy
> - supabase deploy
> Specify when the deploy job should run
> For example, only run on the master branch
> 
> only:
> - master
> 
> In this example, we have defined two stages in the CI/CD pipeline: build and deploy. The build stage is responsible for building your application, and the deploy stage is responsible for deploying your application to Supabase using the Supabase CLI action.
> 
> In the deploy job, we install the Supabase CLI using npm, set the required environment variables for connecting to your Supabase project (i.e., SUPABASE_URL and SUPABASE_KEY), and then run the supabase deploy command to deploy your application to Supabase.
> 
> Note: Make sure to replace <your_supabase_url> and <your_supabase_key> with your actual Supabase URL and API key.
> 
> You can further customize the deployment process by passing additional arguments to the supabase deploy command or by modifying the script section to include any other necessary commands for your specific deployment requirements.
> 
> Additionally, you may need to configure GitLab CI/CD variables to securely store sensitive information, such as the Supabase URL and API key, and use them in your CI/CD pipeline.
> 
> Remember to adjust the GitLab CI/CD pipeline configuration to suit your specific application requirements, Supabase configuration, and deployment process.
> 
> https://amzn.to/3GJj291 Check it this book.

**Reply by AyoPrez** [reply]
I tried this and there is no much information. And as it is, it doesn't work for me. But I found and tried this:

https://www.reddit.com/r/Supabase/comments/1383cg4/ive_created_a_gitlabciyml_file_to_deploy_db/

Hope it helps.
